x/merkledb/trieview.go

```diff
@@ -555,12 +515,18 @@ func (t *trieView) commitChanges(ctx context.Context, trieToCommit *trieView) er
 	return nil
 }
 
-// commitToParent commits the changes from this view to its parent Trie
-// assumes [t.lock] is held
-func (t *trieView) commitToParent(ctx context.Context) error {
-	ctx, span := t.db.tracer.Start(ctx, "MerkleDB.trieview.commitToParent")
+// Commits the changes from [trieToCommit] to this view,
+// this view to its parent, and so on until committing to the db.
+// Assumes [t.db.commitLock] is held.
+func (t *trieView) commitToDB(ctx context.Context) error {
+	ctx, span := t.db.tracer.Start(ctx, "MerkleDB.trieview.commitToDB", oteltrace.WithAttributes(
+		attribute.Int("changeCount", len(t.changes.values)),
+	))
 	defer span.End()
 
+	t.commitLock.Lock()
+	defer t.commitLock.Unlock()
```

Comment:
seems we changed the span to include the time of getting the lock, is this intentional?